### PR TITLE
SWC-5637

### PR DESCRIPTION
**Pull request by nickgros**
Fix SWC-5637

- Fix alignment by switching to MUI
- Use MUI popover. Fixes cases where popovers would not appear over MUI dialogs

The alignment with the form and the dialog isn't perfect, but fixing that properly will require major changes to the SchemaDrivenAnnotationEditor

|Before|After|
|---|---|
|image|image|
|image|image|

### CODE REVIEW

File: packages/synapse-react-client/src/lib/containers/entity/metadata/EntityModal.tsx [hunk 1]
```diff
@@ -60,147 +71,167 @@ export const EntityModal: React.FC<EntityModalProps> = ({
...
-        copy: `Open ${entityTypeToFriendlyName(entityBundle.entityType)}`,
-        onClick: () =>
-          window.open(
-            `${getEndpoint(
-              BackendDestinationEnum.PORTAL_ENDPOINT,
-            )}#!Synapse:${entityId}`,
-            '_blank',
-            'noopener',
-          ),
+  const showOpenEntityPageButton =
```
Comment (nickgros) [inline packages/synapse-react-client/src/lib/containers/entity/metadata/EntityModal.tsx:74]: Way easier to just create the buttons than to try to replicate this `action` -> `Button` mapping

File: packages/synapse-react-client/src/lib/style/components/_annotations.scss [hunk 2]
```diff
@@ -1,156 +1,154 @@
 @use '../abstracts/variables' as SRC;
 
-.bootstrap-4-backport {
```
Comment (nickgros) [inline packages/synapse-react-client/src/lib/style/components/_annotations.scss:3]: Just remove this outer .bootstrap-4-backport class which was unnecessary (and broke styling when I removed it from the modal)

File: packages/synapse-react-client/src/lib/utils/hooks/useOverlay.tsx [hunk 3]
```diff
@@ -70,35 +77,33 @@ export function useOverlay(
 
   const OverlayComponent = useCallback(
     () => (
-      <Overlay
-        target={targetRef.current}
-        show={isShowing}
-        placement={placement}
+      <Popover
```
Comment (nickgros) [inline packages/synapse-react-client/src/lib/utils/hooks/useOverlay.tsx:80]: Use the MUI version. The bootstrap version didn't work in MUI modals

Comment (hallieswan) [reply packages/synapse-react-client/src/lib/utils/hooks/useOverlay.tsx:80]: Nice! Removes another bootstrap-4-backport instance too

**Review by hallieswan — APPROVED** [review]
Looks good! I saw the same issue with the Popover in another MUI modal, so it's great that you've fixed it!

**Pull request merged**